lib/google_maps_geocoder/google_maps_geocoder.rb

```diff
@@ -96,11 +97,28 @@ def http(uri)
 
   def json_from_url(url)
     uri = URI.parse query_url(url)
+
     logger.debug('GoogleMapsGeocoder') { uri }
+
     response = http(uri).request(Net::HTTP::Get.new(uri.request_uri))
     ActiveSupport::JSON.decode response.body
   end
 
+  def handle_exception(google_response_json)
+    status = google_response_json['status']
+    # no exception
+    return if status == 'OK'
+
+    message = GoogleMapsGeocoderErrorHandler::GeneralGoecodingError\
```

Comment:
this module name seems to be too long as well, but having just a `GoogleMapsGeocoder::GeneralGoecodingError` involves some fundamental changes here, so I prefer to wait for your response first.